Crate does not compile on iOS

When trying to build the create on iOS, the build fails with the error:

```
   Compiling fs_at v0.1.7
error[E0425]: cannot find value `O_PATH` in crate `libc`
   --> /Users/felix.barz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/fs_at-0.1.7/src/unix.rs:139:55
    |
139 |             libc::O_RDONLY | libc::O_NOFOLLOW | libc::O_PATH | libc::O_CLOEXEC | libc::O_NOCTTY;
    |                                                       ^^^^^^ not found in `libc`

For more information about this error, try `rustc --explain E0425`.
error: could not compile `fs_at` (lib) due to previous error
```

When checking the sources, whe can see that `open_path_at` is disabled for macos: https://github.com/rbtcollins/fs_at/blob/main/src/unix.rs#L136

This should also be disable for iOS. When adding `target_os = "ios"` to the exclusions (including a few other places), the library compiles again.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -8,7 +8,7 @@ name = "fs_at"
 readme = "README.md"
 repository = "https://github.com/rbtcollins/fs_at.git"
 rust-version = "1.62.0"
-version = "0.1.7"
+version = "0.1.8"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -324,8 +324,8 @@ impl OpenOptions {
     /// Unix: sets O_NOFOLLOW | O_PATH. Many operations on the file handle are
     /// restricted.
     ///
-    /// AIX, MacOSX and NetBSD: Not implemented as O_PATH is not defined.
-    #[cfg(not(any(target_os = "aix", target_os = "macos", target_os = "netbsd")))]
+    /// AIX, MacOSX, iOS and NetBSD: Not implemented as O_PATH is not defined.
+    #[cfg(not(any(target_os = "aix", target_os = "macos", target_os = "ios", target_os = "netbsd")))]
     pub fn open_path_at<P: AsRef<Path>>(&self, d: &File, p: P) -> Result<File> {
         self._impl
             .open_path_at(d, OpenOptions::ensure_rootless(p.as_ref())?)
@@ -521,7 +521,7 @@ pub mod testsupport;
 
 #[cfg(test)]
 mod tests {
-    #[cfg(all(not(target_os = "macos"), not(target_os = "netbsd")))]
+    #[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "netbsd")))]
     use std::path::Path;
     use std::{
         ffi::OsStr,
@@ -1216,7 +1216,7 @@ mod tests {
         Ok(())
     }
 
-    #[cfg(not(any(target_os = "aix", target_os = "macos", target_os = "netbsd")))]
+    #[cfg(not(any(target_os = "aix", target_os = "macos", target_os = "ios", target_os = "netbsd")))]
     #[test]
     fn open_path_at() -> Result<()> {
         let (_tmp, parent_dir, _pathname) = setup()?;

--- a/src/unix.rs
+++ b/src/unix.rs
@@ -133,7 +133,7 @@ impl OpenOptionsImpl {
         self._open_at(d, path, flags)
     }
 
-    #[cfg(not(any(target_os = "aix", target_os = "macos", target_os = "netbsd")))]
+    #[cfg(not(any(target_os = "aix", target_os = "macos", target_os = "ios", target_os = "netbsd")))]
     pub fn open_path_at(&self, d: &File, path: &Path) -> Result<File> {
         let flags =
             libc::O_RDONLY | libc::O_NOFOLLOW | libc::O_PATH | libc::O_CLOEXEC | libc::O_NOCTTY;

